mod_nested_tooltips/hooks/ui/screens/tooltip/tooltip_events.nut

```diff
@@ -80,7 +87,7 @@
 
 		if (ret == null)
 		{
-			skill = ::MSU.NestedTooltips.SkillObjectsByFilename[_data.Filename];
+			skill = ::new(scriptPath);
```

Comment:
So we do that caching to items but not skills, why?

Reply:
Doesn't seem necessary and just hogs memory. For items it is _necessary_ because we need to have the item available for the nested tooltips of skills which want to refer to that item e.g. reload_bolt which needs an item.

Reply:
We do that in this new system, we create it on demand. But we need it to persist outside this scope as well because the skill's tooltip will need to refer to that item. If we just create it on demand and return it, the object will be garbage collected and we won't have access to it anymore. See how our hook in `tooltip_events.nut` where we need to access an item if necessary, so the item needs to exist.